docs/cli/ilab-rag-retrieval.md

```diff
@@ -0,0 +1,221 @@
+│  • /lc: list contexts                                                                                                             │
+│  • /m: toggle multiline (for the next session only)                                                                               │
+│  • /M: toggle multiline                                                                                                           │
+│  • /n: new session                                                                                                                │
+│  • /N: new session (ignoring loaded)                                                                                              │
+│  • /d <int>: display previous response based on input, if passed 1 then previous, if 2 then second last response and so on.       │
+│  • /p <int>: previous response in plain text based on input, if passed 1 then previous, if 2 then second last response and so on. │
+│  • /r: toggle the status of the RAG pipeline.                                                                                     │
+│  • /md <int>: previous response in Markdown based on input, if passed 1 then previous, if 2 then second last response and so on.  │
+│  • /s filepath: save current session to filepath                                                                                  │
+│  • /l filepath: load filepath and start a new session                                                                             │
+│  • /L filepath: load filepath (permanently) and start a new session                                                               │
+│                                                                                                                                   │
+│ Press Alt (or Meta) and Enter or Esc Enter to end multiline input.                                                                │
+╰───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
+```
+
+### 2.6 RAG Chat Options
+As we stated in [3.1 Working Assumptions](#31-working-assumption), we will introduce new configuration options for the spceific `chat` command,
+but we'll use flags and environment variables for the options that come from the embedding ingestion pipeline command.
+
+| Configuration FQN | Description | Default Value | CLI Flag | Environment Variable |
+|-------------------|-------------|---------------|----------|----------------------|
+| chat.rag.enabled | Enable or disable the RAG pipeline. | `false` | `--rag` (boolean)| `ILAB_CHAT_RAG_ENABLED` |
+| chat.rag.retriever.top_k | The maximum number of documents to retrieve. | `10` | `--retriever-top-k` | `ILAB_CHAT_RAG_RETRIEVER_TOP_K` |
+| chat.rag.prompt | Prompt template for RAG-based queries. | Examples below | `--rag-prompt` | `ILAB_CHAT_RAG_PROMPT` |
```

Comment:
- Would this deserve its own ADR?
- Is it something we should let the user to configure or are you just thinking of where to place an hardcoded prompt?

Reply:
Yes, I will add this to the list of planned ADRs.

Reply:
> If there is a way to unify prompt templates throughout InstructLab into one place rather than adding another place to store them, that would probably be ideal.

+1